src/main/java/com/okta/tools/awscli.java

```diff
@@ -22,15 +22,28 @@
 
 public class awscli {
     public static void main(String[] args) throws Exception {
+        // support named profiles from the CLI
+        // https://docs.aws.amazon.com/cli/latest/userguide/cli-multiple-profiles.html
+        String profile = null;
+        boolean hasProfile = false;
+        for (String arg : args) {
+            if (hasProfile) {
+                profile = arg;
+                break;
+            }
+            hasProfile = "--profile".equals(arg);
+        }
```

Comment:
👍 This is a clever command-line switch handling algorithm. It took me a minute to figure it out, but I like it.

❓ I'm curious. How do you handle those duplicate args in the downstream command? Running `./awcli --profile prod sts get-caller-identity` is going to end up making a sub-process which is equivalent to running `aws --profile prod --profile prod sts get-caller-identity`. Do you anticipate any problems with that? I think you might find calling **WithOkta.main** when **--profile** is supplied to be more closely aligned with your use case.

Reply:
currently awscli adds --profile <profile> to the command line.  This because I'm processing --profile with the highest priority, the profile strings will always match, and it will just be specified twice on the command line.  I chose to not try and eliminate the duplicated in order to keep the code simple.
i.e awscli s3 ls s3://mybucket --profile XXX   will spawn aws s3 ls s3://mybucket --profile XXX --profile XXX 

which is a no-op